src/CodeGen_LLVM.cpp

```diff
@@ -555,8 +556,11 @@ void CodeGen_LLVM::compile_func(const LoweredFunc &f) {
     // If the Func is externally visible, also create the argv wrapper
     // (useful for calling from JIT and other machine interfaces).
     if (f.linkage == LoweredFunc::External) {
-        add_argv_wrapper(module, function, name + "_argv");
-        embed_metadata(name + "_metadata", args);
+        llvm::Function *wrapper = add_argv_wrapper(module, function, name + "_argv");
+        llvm::Constant *metadata = embed_metadata(name + "_metadata", args);
```

Comment:
Before this PR, metadata is always emitted, but not referenced by anything, so it's subject to dead-stripping if not used.

After this PR, metadata is always emitted, but still not referenced by default; if you build with RegisterMetadata enabled, the metadata is then always referenced and never subject to dead-stripping.

When we landed the first metadata PR we decided to always emit it because there's no real downside (ie, it's probably going to be dead-stripped if unused); I think that's still the right thing.

RegisterMetadata is trickier, since code that doesn't use it really doesn't want this data present. (Ideally there would be a way to emit the registration code at all times, but allow it to be deadstripped if the final linkage never called the enumerate() function; I haven't been able to think of a way to enable that, unfortunately.)